### PR TITLE
modbus 8e1 support + mb options to menuconfig

### PR DESCRIPTION
**Pull request by alezz**
This is a small patch to fully support RTU modbus.
 * Added 8e1 support (which is the default port config in RTU modbus standard) to usart.h;
 * Added modbus options (port baud and parity/bit settings) to the menuconfig;
 * Added RS485TE pin definition in netio.m4

### CODE REVIEW

File: pinning/hardware/netio.m4 [hunk 1]
```diff
@@ -1,3 +1,6 @@
+/* add support for RS485 */
+pin(RS485TE_USART0, PD2, OUTPUT)
```
Comment (michaelb42) [inline pinning/hardware/netio.m4]: PD2 is already used by multiple other modules. Please enclose this definition in a conditional.

File: core/usart.h [hunk 2]
```diff
@@ -135,6 +135,26 @@ usart_init(void) \
...
+static void \
+usart_init(void) \
+{\
+  /* The ATmega644 datasheet suggests to clear the global\
+   * interrupt flags on initialization ... */\
+  ATOMIC_BLOCK(ATOMIC_RESTORESTATE) \
+  { \
+    usart(UBRR,H) = UBRRH_VALUE; \
+    usart(UBRR,L) = UBRRL_VALUE; \
+    /* set mode 8N1: 8 bits, 1 stop, even parity, asynchronous usart \
```
Comment (michaelb42) [inline core/usart.h]: suggest 8E1 instead of 8N1 in comment

**Comment by alezz** [comment]
The last commit:
- 485TE pinning (conditional ifdef 'conf_MODBUS' and using PC0 that should be less used in any case);
- 8e1 comment ok;
- created cleaner submenu for menuconfig, and added a comment to remember the user to check for 485TE pinning.

File: pinning/hardware/netio.m4 [hunk 3]
```diff
@@ -1,5 +1,7 @@
-/* add support for RS485 */
-pin(RS485TE_USART0, PD2, OUTPUT)
+ifdef(`conf_MODBUS', `dnl
+  /* add support for RS485 */
+  pin(RS485TE_USART0, PC0, OUTPUT)
```
Comment (michaelb42) [inline pinning/hardware/netio.m4]: Uh, PC0 is I2C SCL - not a very good choice. But anyway, users have to adapt this to their own hardware.

Comment (alezz) [inline pinning/hardware/netio.m4]: Hmm you're right, maybe better PD3 or PD2 that at least are near TXD/RXD pins. But yes they have to customize that, I have added the comment to menuconfig just for that reason.

**Comment by eku** [comment]
New options for menuconfig **must** provide help.

**Comment by alezz** [comment]
@eku 
- i don't see the need of help for simple options, in fact there is no help for "modbus uart select" that was already there. However I can provide help, if you suggest me where to write it as it's not in config.in from what I can see.
- scripts, regenerate-profiles && autotest : I must say I do not know anything about them, is there a documentation I can read about?
- pinning: if the project policy is not to add hardware that is not provided/expected with the original hardware, I will discard changes to that file.

**Comment by eku** [comment]
@alezz 

- menuconfig provides a way to add documentation to doc/Configure.help. Just select ```Help``` and enter editor.
- New options tend to break existing profiles and therefor autotest.
- Yes, the pinning for existing hardware should carry only original pinnings. That's why I made you aware of ```add-hardware```.

**Comment by alezz** [comment]
@eku 
i have added help entries, entry in add-hardware, reverted netio.m4 to original state, done regenerate-profiles and autotest. Should be okay now.

**Pull request merged**